src/app/config/setup/octoprint-basic-auth/octoprint-basic-auth.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { BasicAuth } from '../../config.model';
+
+@Component({
+  selector: 'app-config-setup-octoprint-basic-auth',
+  templateUrl: './octoprint-basic-auth.component.html',
+  styleUrls: ['./octoprint-basic-auth.component.scss', '../setup.component.scss'],
+})
+export class OctoprintBasicAuthComponent {
+  @Input() basicAuth: BasicAuth = {};
+
+  @Output() basicAuthChange = new EventEmitter<BasicAuth>();
+
+  get user(): string {
+    return this.basicAuth && this.basicAuth.user;
+  }
+
+  set user(user: string) {
+    if (!this.basicAuth) {
+      this.basicAuth = {};
+    }
+    this.basicAuth.user = user;
+  }
+
+  get pass(): string {
+    return this.basicAuth && this.basicAuth.pass;
```

Comment:
Doesn't that just return true or false, based on whether the objects are available? Another guy showed me https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining in a PR the other day, might be something worth considering here (if it makes sense).